app/services/dual-output/dual-output.ts

```diff
@@ -360,80 +381,291 @@ export class DualOutputService extends PersistentStatefulService<IDualOutputServ
+      this.convertSingleOutputToDualOutputCollection();
     }
+    if (this.state.isLoading) this.SET_IS_LOADING(false);
+  }
+
+  validateDualOutputCollection() {
+    this.SET_IS_LOADING(true);
+    // confirm dual output scene collection
+    this.scenesService.views.scenes.forEach(scene => {
+      if (this.views.sceneNodeMaps[scene.id]) {
+        this.confirmSceneNodeMap(scene.id, this.views.sceneNodeMaps[scene.id]);
+      } else {
+        this.createVerticalNodes(scene.id);
+      }
+    });
+    this.SET_IS_LOADING(false);
   }
 
-  convertSceneSources(sceneId: string) {
-    const sceneSources = this.scenesService.views.sceneSourcesForScene(sceneId);
-    if (sceneSources.length > 0) {
-      sceneSources.forEach(scene => this.confirmOrCreateVerticalNodes(scene.sourceId));
+  convertSingleOutputToDualOutputCollection() {
+    // if in dual output mode, convert the single output collection to dual output
+    this.scenesService.views.scenes.forEach(scene => {
+      this.createVerticalNodes(scene.id);
+    });
+  }
+
+  /**
+   * Create a vertical node to partner with the vertical node
+   * @param horizontalNode - Node to copy to the vertical display
+   *
+   * @remark The horizontal node id is always the key in the scene node map.
+   * The node map entry is so that the horizontal and vertical nodes can refer to each other.
+   */
+  createVerticalNode(horizontalNode: TSceneNode, verticalNodeId?: string): TSceneNode {
+    const scene = horizontalNode.getScene();
+
+    if (horizontalNode.isFolder()) {
+      // add folder and create node map entry
+      const folder = scene.createFolder(horizontalNode.name, {
+        id: verticalNodeId,
+        display: 'vertical',
+      });
+      this.sceneCollectionsService.createNodeMapEntry(scene.id, horizontalNode.id, folder.id);
+
+      this.sceneNodeHandled.next();
+      return folder;
+    } else {
+      // add item
+      const item = scene.addSource(horizontalNode.sourceId, {
+        id: verticalNodeId,
+        display: 'vertical',
+      });
+
+      // position all of the nodes in the upper left corner of the vertical display
+      // so that all of the sources are visible
+      item.setTransform({ position: { x: 0, y: 0 } });
+
+      // show all vertical scene items by default
+      item.setVisibility(true);
+
+      // match locked
+      item.setLocked(horizontalNode.locked);
+
+      this.sceneCollectionsService.createNodeMapEntry(scene.id, horizontalNode.id, item.id);
+
+      this.sceneNodeHandled.next();
+      return item;
     }
   }
 
   /**
-   * Assign or confirm node contexts to a dual output scene
-   * @param sceneId - Id of the scene to map
+   * Create a horizontal node to partner with the vertical node
+   * @param verticalNode - Node to copy to the horizontal display
+   *
+   * @remark The horizontal node id is always the key in the scene node map.
+   * The node map entry is so that the horizontal and vertical nodes can refer to each other.
```

Comment:
in which situations are we expecting to have vertical nodes without a corresponding horizontal node?

Reply:
Right now, there should never be a vertical node without a corresponding horizontal node. If there is for whatever reason, when the collection is loaded, the repair function will create a horizontal node to partner with the vertical node. In the scene source fix, this will introduce the instance where there is a vertical node created without a horizontal one. But this is because the scene source fix creates a vertical scene that is only ever rendered as a source for a scene-as-scene-item. More on that to come.